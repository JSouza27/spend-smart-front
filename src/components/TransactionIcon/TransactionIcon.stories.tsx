import { ComponentMeta, ComponentStory } from '@storybook/react'
import { ThemeProvider } from 'styled-components';

import TransactionIcon from '.'
import theme from '../../styles/theme';

export default {
  title: 'TransactionIcon',
  component: TransactionIcon
} as ComponentMeta<typeof TransactionIcon>

export const Basic: ComponentStory<typeof TransactionIcon> = () => (
  <ThemeProvider theme={theme}>
    <TransactionIcon />
  </ThemeProvider>
)
