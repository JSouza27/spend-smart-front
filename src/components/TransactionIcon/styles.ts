import styled, { DefaultTheme, css } from 'styled-components';

import { TransactionIconProps } from '.';

type WrapperProps = Pick<TransactionIconProps, 'type'>;

const wrapperModifiers = {
  income: (theme: DefaultTheme) => css`
    background-color: ${theme.colors.green_400};
  `,
  expense: (theme: DefaultTheme) => css`
    background-color: ${theme.colors.red_400};
    transform: scale(-1) rotate(270deg);
  `
};
export const Wrapper = styled.div<WrapperProps>`
  ${({ theme, type }) => css`
    align-items: center;
    border-radius: 100%;
    color: ${theme.colors.neutral_0};
    display: flex;
    justify-content: center;
    height: 3.2rem;
    overflow: hidden;
    width: 3.2rem;

    ${!!type && wrapperModifiers[type](theme)}
  `}
`;
