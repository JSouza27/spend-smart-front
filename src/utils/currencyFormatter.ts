export class CurrencyFormatter {
  private static language = 'pt-BR';
  private static currency = 'BRL';

  static formatter(num: number): string {
    return (num ?? 0).toLocaleString(this.language, {
      currency: this.currency
    });
  }

  static unformat(num: string): number {
    const nums = num.replace(/[^0-9]+/g, '');
    const i = nums.length - 2;

    return Number(`${nums.substring(0, i)}.${nums.substring(i)}`);
  }
}
