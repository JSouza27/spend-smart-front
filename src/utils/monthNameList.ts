export const monthNameList = [
  'janeiro',
  'fevereiro',
  'mar√ßo',
  'abril',
  'maio',
  'junho',
  'julho',
  'agosto',
  'setembro',
  'outubro',
  'novembro',
  'dezembro'
];
