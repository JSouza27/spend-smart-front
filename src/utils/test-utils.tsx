import { render } from '@testing-library/react';
import React from 'react';
import { ThemeProvider } from 'styled-components';

import theme from '../styles/theme';
import { FormProvider, useForm } from 'react-hook-form';

const AllTheProviders = ({ children }: any) => {
  const methods = useForm();

  return (
    <ThemeProvider theme={theme}>
      <FormProvider {...methods}>{children}</FormProvider>
    </ThemeProvider>
  );
};

const customRender = (
  ui: React.ReactElement<any, string | React.JSXElementConstructor<any>>,
  options?: any
) => render(ui, { wrapper: AllTheProviders, ...options });

export * from '@testing-library/react';
export { customRender as render };
