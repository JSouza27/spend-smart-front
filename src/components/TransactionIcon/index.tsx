import { ArrowTrending } from '@styled-icons/fluentui-system-filled';

import * as S from './styles';

export type TransactionIconProps = {
  type: 'income' | 'expense';
};

export default function TransactionIcon({ type }: TransactionIconProps) {
  return (
    <S.Wrapper data-testid="arrow-icon" type={type}>
      <ArrowTrending className="icon" size={16} title="icon" />
    </S.Wrapper>
  );
}
