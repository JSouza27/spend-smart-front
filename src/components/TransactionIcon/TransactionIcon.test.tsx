import { render, screen } from 'utils/test-utils';

import TransactionIcon from '.';

describe('<TransactionIcon />', () => {
  it('should have the color #D14343', () => {
    const { container } = render(<TransactionIcon type="expense" />);

    const icon = screen.getByTestId('arrow-icon');
    expect(icon).toBeInTheDocument();
    expect(icon).toHaveStyle('background-color: #D14343');
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should have the color #52BD94', () => {
    const { container } = render(<TransactionIcon type="income" />);

    const icon = screen.getByTestId('arrow-icon');
    expect(icon).toBeInTheDocument();
    expect(icon).toHaveStyle('background-color: #52BD94');
    expect(container.firstChild).toMatchSnapshot();
  });
});
