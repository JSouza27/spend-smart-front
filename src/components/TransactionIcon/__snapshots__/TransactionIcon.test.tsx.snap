// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TransactionIcon /> should have the color #52BD94 1`] = `
.c1 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 100%;
  color: #FFFFFF;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 3.2rem;
  overflow: hidden;
  width: 3.2rem;
  background-color: #52BD94;
}

<div
  class="c0"
  data-testid="arrow-icon"
  type="income"
>
  <svg
    class="c1 icon"
    fill="currentColor"
    focusable="false"
    height="16"
    role="img"
    viewBox="0 0 24 24"
    width="16"
    xmlns="http://www.w3.org/2000/svg"
  >
    <title>
      icon
    </title>
    <path
      d="M14 5.5h7a1 1 0 0 1 1 .88v7.12a1 1 0 0 1-2 .12V8.9l-7.3 7.3a1 1 0 0 1-1.31.08l-.1-.08L9 13.9l-5.28 5.3a1 1 0 0 1-1.5-1.32l.08-.1 6-6a1 1 0 0 1 1.32-.08l.1.08L12 14.1l6.58-6.59H14a1 1 0 0 1-.99-.88V6.5a1 1 0 0 1 .88-1H21h-7Z"
    />
  </svg>
</div>
`;

exports[`<TransactionIcon /> should have the color #D14343 1`] = `
.c1 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 100%;
  color: #FFFFFF;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 3.2rem;
  overflow: hidden;
  width: 3.2rem;
  background-color: #D14343;
  -webkit-transform: scale(-1) rotate(270deg);
  -ms-transform: scale(-1) rotate(270deg);
  transform: scale(-1) rotate(270deg);
}

<div
  class="c0"
  data-testid="arrow-icon"
  type="expense"
>
  <svg
    class="c1 icon"
    fill="currentColor"
    focusable="false"
    height="16"
    role="img"
    viewBox="0 0 24 24"
    width="16"
    xmlns="http://www.w3.org/2000/svg"
  >
    <title>
      icon
    </title>
    <path
      d="M14 5.5h7a1 1 0 0 1 1 .88v7.12a1 1 0 0 1-2 .12V8.9l-7.3 7.3a1 1 0 0 1-1.31.08l-.1-.08L9 13.9l-5.28 5.3a1 1 0 0 1-1.5-1.32l.08-.1 6-6a1 1 0 0 1 1.32-.08l.1.08L12 14.1l6.58-6.59H14a1 1 0 0 1-.99-.88V6.5a1 1 0 0 1 .88-1H21h-7Z"
    />
  </svg>
</div>
`;
